controllers/C_Document.class.php

```diff
@@ -612,75 +650,88 @@ function retrieve_action($patient_id = "", $document_id, $as_file = true, $origi
+                //try to convert it if it has not yet been converted
+                $originalUrl = $url;
                 $convertedFile = substr(basename_international($url), 0, strrpos(basename_international($url), '.')) . '_converted.jpg';
-                if ($couch_docid && $couch_revid) {
-                    $url = $GLOBALS['OE_SITE_DIR'] . '/documents/temp/' . $convertedFile;
+                $url = $GLOBALS['OE_SITE_DIR'] . '/documents/' . $from_pathname . '/' . $convertedFile;
+                if (!is_file($url)) {
+                    if ($d->get_encrypted() == 1) {
+                        // decrypt the from-file into a temporary file
+                        $from_file_unencrypted = decryptStandard(file_get_contents($originalUrl), null, 'database');
+                        $from_file_tmp_name = tempnam($GLOBALS['temporary_files_dir'], "oer");
+                        file_put_contents($from_file_tmp_name, $from_file_unencrypted);
+                        // prepare a temporary file for the unencrypted to-file
+                        $to_file_tmp_name = tempnam($GLOBALS['temporary_files_dir'], "oer") . ".jpg";
+                        // convert file to jpg
+                        exec("convert -density 200 " . escapeshellarg($from_file_tmp_name) . " -append -resize 850 " . escapeshellarg($to_file_tmp_name));
+                        // remove unencrypted tmp file
+                        unlink($from_file_tmp_name);
+                        // make the encrypted to-file if a to-file was created in above convert call
+                        if (is_file($to_file_tmp_name)) {
+                            $to_file_encrypted = encryptStandard(file_get_contents($to_file_tmp_name), null, 'database');
+                            file_put_contents($url, $to_file_encrypted);
+                            // remove unencrypted tmp file
+                            unlink($to_file_tmp_name);
+                        }
+                    } else {
+                        // convert file to jpg
+                        exec("convert -density 200 " . escapeshellarg($originalUrl) . " -append -resize 850 " . escapeshellarg($url));
+                    }
+                }
+                if (is_file($url)) {
+                    if ($d->get_encrypted() == 1) {
+                        $filetext = decryptStandard(file_get_contents($url), null, 'database');
+                    } else {
+                        $filetext = file_get_contents($url);
+                    }
                 } else {
-                    $url = $GLOBALS['OE_SITE_DIR'] . '/documents/' . $from_pathname . '/' . $convertedFile;
+                    $filetext = '';
+                    echo "<b>NOTE</b>: " . xlt('Document') . "'" . text($fname) . "' " . xlt('cannot be converted to JPEG. Perhaps ImageMagick is not installed?') . "<br><br>";
```

Comment:
TODO
Likely need to deal with this message differently now that brought in documents class